sdk-rs/src/utils.rs

```diff
@@ -39,6 +40,27 @@ pub fn load_keypair_multi_format(path_or_key: &str) -> SdkResult<Keypair> {
     }
 }
 
+pub fn http_to_ws(url: &str) -> Result<String, &'static str> {
+    let base_url = if url.starts_with("http://") {
+        url.replacen("http://", "ws://", 1)
+    } else if url.starts_with("https://") {
+        url.replacen("https://", "wss://", 1)
+    } else {
+        return Err("Invalid URL scheme");
+    };
+
+    Ok(format!("{}/ws", base_url.trim_end_matches('/')))
+}
+
+pub fn to_ws_json(config: &dyn MarketConfig) -> String {
```

Comment:
can do static dispatch here
```suggestion
pub fn to_ws_json(config: impl MarketConfig) -> String {
```